components/orders/PaymentMethodPicker.tsx

```diff
@@ -0,0 +1,440 @@
+import MessageBoxGraphqlError from '../MessageBoxGraphqlError';
+
+export const paymentMethodPickerQuery = gql`
+  query PaymentMethodPicker($accountSlug: String!, $hostSlug: String!) {
+    account(slug: $accountSlug) {
+      id
+      name
+      paymentMethods(type: [CREDITCARD, US_BANK_ACCOUNT, SEPA_DEBIT, BACS_DEBIT, GIFTCARD, PREPAID, COLLECTIVE]) {
+        id
+        name
+        data
+        service
+        type
+        account {
+          id
+          name
+          slug
+        }
+        balance {
+          value
+          valueInCents
+          currency
+        }
+        limitedToHosts {
+          id
+        }
+        sourcePaymentMethod {
+          id
+          limitedToHosts {
+            id
+          }
+        }
+      }
+    }
+    host(slug: $hostSlug) {
+      id
+      paypalClientId
+      supportedPaymentMethods
+    }
+  }
+`;
+
+type PaymentMethodPickerProps = {
+  className?: string;
+  value: PaymentMethodOption;
+  onChange: (option: PaymentMethodOption) => void;
+  host: Pick<Host, 'slug'>;
+  account: Pick<Account, 'slug'>;
+  order?: Order;
+};
+
+export type PaymentMethodOption =
+  | {
+      id: 'stripe-payment-element';
+      stripe?: Stripe;
+      elements?: StripeElements;
+      paymentElement?: StripePaymentElement;
+      setupIntent?: StripeSetupIntent;
+    }
+  | {
+      id: 'pay-with-paypal';
+    }
+  | {
+      id: string;
+      name?: string;
+      type?: PaymentMethodType;
+    };
+
+export default function PaymentMethodPicker(props: PaymentMethodPickerProps) {
+  const [error, setError] = React.useState();
+  const query = useQuery<{
+    account: Pick<Account, 'id' | 'name' | 'paymentMethods'>;
+    host: Pick<Host, 'id' | 'paypalClientId' | 'supportedPaymentMethods'>;
+  }>(paymentMethodPickerQuery, {
+    context: API_V2_CONTEXT,
+    variables: {
+      hostSlug: props.host?.slug,
+      accountSlug: props.account?.slug,
+    },
+    skip: !props.host || !props.account,
+    fetchPolicy: 'network-only',
+  });
+
+  const hostHasStripe = query.data?.host?.supportedPaymentMethods?.includes(PaymentMethodLegacyType.CREDIT_CARD);
+
+  const savedMethods = React.useMemo(() => {
+    if (!query.data?.account?.paymentMethods) {
+      return [];
+    }
+
+    return query.data.account.paymentMethods.filter(
+      paymentMethodFilter({
+        host: query.data.host,
+      }),
+    );
+  }, [query?.data?.host, query?.data?.account?.paymentMethods]);
+
+  const onChange = React.useCallback(
+    (option: PaymentMethodOption) => {
+      props.onChange(option);
+    },
+    [props.onChange],
+  );
+
+  if (query.loading) {
+    return <Loading />;
+  }
+
+  if (query.error || error) {
+    return <MessageBoxGraphqlError error={query.error || error} />;
+  }
+
+  return (
+    <form action="#">
+      <fieldset className={clsx(props.className)}>
+        <div className="flex flex-col gap-3">
+          {hostHasStripe && (
+            <StripeSetupPaymentMethodOption
```

Comment:
Not sure if this is a bug of the test environment, but I was able to create a SEPA payment method for a USD payment. Which latter could not be charged by the recurring contributions script.

```
npm run script cron/hourly/00-charge-recurring-contributions.js

> script
> babel-node --extensions .js,.ts $1 cron/hourly/00-charge-recurring-contributions.js

info: Connecting to postgres://127.0.0.1/opencollective_dvl
info: Fixer API is not configured, lib/currency will always return 1.1
1 recurring contributions pending charges. Charging 1 contributions right now. dryRun: false
info: order: 6338, subscription: 4192, attempt: #0, due: Mon Dec 04 2023 00:00:00 GMT+0100 (Central European Standard Time)
error: The currency provided (usd) is invalid. Payments with sepa_debit support the following currencies: eur. {"requestId":"req_3dlNNeTSrIkdw6","statusCode":400}
Writing the output to a CSV file
Sending email report
Finished running charge recurring contributions
```

Reply:
Handled in the latest push by limiting the stripe payment element options and the saved methods list.